File: PRS web/PRS web/Models/PurchaseRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace PRS_web.Models
{
    public class PurchaseRequest
    {
        public int Id { get; set; }
        public virtual User UserId { get; set; }
        public virtual User User { get; set; }
        [StringLength(100)]
        [Required]
        public string Description { get; set; }
        [StringLength(255)]
        [Required]
        public string Justification { get; set; }
        public DateTime DateNeeded { get; set; }
        [StringLength(25)]
        [Required]
        public string DeliveryMode { get; set; }
        [Required]
        public int StatusId { get; set; }
        [Required]
        public decimal Total { get; set; }
        [Required]
        public  DateTime SubmittedDate { get; set; }
        [Required]
        public bool Active { get; set; }
        [Required]
        [StringLength(100)]
        public string ReasonForRejection { get; set; }
        public DateTime DateCreated { get; set; }
        public DateTime DateUpdated { get; set; }
}
}

File: PRSmaster/PRSweb-master/PRSweb-master/ng/src/app/models/PurchaseRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var PurchaseRequest = /** @class */ (function () {
    function PurchaseRequest(ID, Description, Justification, DateNeeded, DeliveryMode, Status, Total, SubmittedDate, UserId) {
        this.ID = ID;
        this.Description = Description;
        this.Justification = Justification;
        this.DateNeeded = DateNeeded;
        this.DeliveryMode = DeliveryMode;
        this.Status = Status;
        this.Total = Total;
        this.SubmittedDate = SubmittedDate;
        this.UserId = UserId;
    }
    return PurchaseRequest;
}());
exports.PurchaseRequest = PurchaseRequest;
